sk-typescript-console-chat/consoleChat.ts

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "./bin/Microsoft.SemanticKernel.Core.js";
+import "./bin/Microsoft.SemanticKernel.Connectors.AI.OpenAI.js";
+import dotnet from "node-api-dotnet";
+import readline from "node:readline";
+import { consoleColours } from "./consoleColours.js";
+
+const SK = dotnet.Microsoft.SemanticKernel;
+
+// The JS marshaller does not yet support extension methods.
+const kernelBuilder = SK.OpenAIKernelBuilderExtensions.WithAzureChatCompletionService(
+    SK.Kernel.Builder,
+    process.env['OPENAI_DEPLOYMENT'] || '',
+    process.env['OPENAI_ENDPOINT'] || '',
+    process.env['OPENAI_KEY'] || '',
+);
+
+const kernel = kernelBuilder
+    .Build();
+
+const r1 = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+    prompt: 'SK> Hello. Ask me a question or say goodbye to exit.'
+});
+
+let chatHistory: string[] = [];
+
+r1.prompt();
+console.log("");
+r1.on('line', async (userInput: string) => {
+    if (userInput.toLowerCase() === 'goodbye') {
+        console.log('SK> Goodbye!');
+        process.exit(0);
+    } else {
+        chatHistory.push(userInput);
+        // The JS marshaller does not yet support extension methods.
+        const chatFunction = SK.InlineFunctionsDefinitionExtension
+            .CreateSemanticFunction(kernel, chatHistory.join(""));
```

Comment:
We can revisit this when I figure out how to get the template engine to work as creating a new semantic function every time isn't a good pattern to promote.